File: src/Test.CSharp.WinForms/VkontakteFriends.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Collections.Specialized;
using Nemiro.OAuth;

namespace Test.CSharp.WinForms
{
  public partial class VkontakteFriends : Form
  {
    public VkontakteFriends()
    {
      InitializeComponent();
    }

    public VkontakteFriends(string accessToken)
    {
      InitializeComponent();

      // query parameters
      var parameters = new NameValueCollection
      { 
        { "access_token", accessToken },
        { "count", "10" }, // first 10 friends
        { "fields", "nickname" }
      };

      // execute the request
      var result = Nemiro.OAuth.OAuthUtility.ExecuteRequest
      (
        "GET",
        "https://api.vk.com/method/friends.get",
        parameters,
        null
      );

      foreach (UniValue itm in result["response"])
      {
        string friendName = "";
        if (itm.ContainsKey("first_name") && itm.ContainsKey("last_name"))
        {
          friendName = String.Format("{0} {1}", itm["first_name"], itm["last_name"]);
        }
        else
        {
          friendName = itm["nickname"].ToString();
        }
        listBox1.Items.Add(friendName);
      }
      
    }
  }
}

File: examples/Test/Test.VB.WinForms/VkontakteFriends.vb
Imports System.Collections.Specialized
Imports Nemiro.OAuth

Public Class VkontakteFriends

  Public Sub New(accessToken As String)
    InitializeComponent()

    ' get user friends
    ' query parameters
    Dim parameters As New NameValueCollection() From _
    {
      {"access_token", accessToken},
      {"count", "10"},
      {"fields", "nickname"}
    }

    ' execute the request
    Dim result As RequestResult = OAuthUtility.ExecuteRequest _
    (
      "GET",
      "https://api.vk.com/method/friends.get",
      parameters,
      Nothing
    )

    For Each itm As UniValue In result("response")
      Dim friendName As String = ""
      If itm.ContainsKey("first_name") AndAlso itm.ContainsKey("last_name") Then
        friendName = String.Format("{0} {1}", itm("first_name"), itm("last_name"))
      Else
        friendName = itm("nickname").ToString()
      End If
      listBox1.Items.Add(friendName)
    Next
  End Sub

End Class